### PR TITLE
chore: use list instead of map for Provider Resources

### PR DESCRIPTION
**Pull request by arkodg**
* list adds order stability

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]

**Comment by cnvergence** [comment]
tests need to be updated

File: internal/message/types.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 type ProviderResources struct {
 	// GatewayAPIResources is a map from a GatewayClass name to
 	// a group of gateway API and other related resources.
-	GatewayAPIResources watchable.Map[string, *gatewayapi.GatewayClassResources]
```
Comment (cnvergence) [inline internal/message/types.go:24]: *gatewayapi.Resources will also need a generated DeepCopy functions

File: internal/gatewayapi/resource.go [hunk 2]
```diff
@@ -20,41 +20,6 @@ import (
 
 type XdsIRMap map[string]*ir.Xds
 type InfraIRMap map[string]*ir.Infra
-type GatewayClassResources map[string]*Resources
```
Comment (cnvergence) [inline internal/gatewayapi/resource.go:23]: list will hit a similar issue
```bash
panic: watchable.DeepCopy: type is not copiable: *[]*gatewayapi.Resources
```

**Comment by Xunzhuo** [comment]
@arkodg conflicts need to be resolved.

File: internal/gatewayapi/resource.go (hunk 2 above)
Comment (arkodg) [reply internal/gatewayapi/resource.go:23]: yeah I had to write my own `DeepCopy` as well

File: internal/gatewayapi/resource.go [hunk 3]
```diff
@@ -20,41 +24,6 @@ import (
 
 type XdsIRMap map[string]*ir.Xds
```
Comment (Xunzhuo) [inline internal/gatewayapi/resource.go:25]: Should we also use list for xdsIR infraIR instead of map

Comment (arkodg) [reply internal/gatewayapi/resource.go:25]: thats an internal structure, iirc each resource is unrelated and the map improves lookup

Comment (arkodg) [reply internal/gatewayapi/resource.go:25]: https://github.com/envoyproxy/gateway/blob/3e70498b63803d8bd78dcad31883bd7211cda3f8/internal/ir/xds.go#L87 helps deal with out of order equality when comparing ir xds

**Review by zirain — APPROVED** [review]
LGTM

**Pull request merged**